core-bundle/src/Security/TwoFactor/Authenticator.php

```diff
@@ -31,6 +31,8 @@ class Authenticator
      */
     public function validateCode(User $user, string $code, int|null $timestamp = null): bool
     {
+        $this->now();
+
         $totp = TOTP::create($this->getUpperUnpaddedSecretForUser($user), clock: $this->clock);
```

Comment:
```suggestion
        $totp = TOTP::create($this->getUpperUnpaddedSecretForUser($user), clock: $this->now());
```

Reply:
That cannot work? It requires a `ClockInterface`, not a `DateTimeImmutable`.

Reply:
That won't work as $this->now returns \DateTimeImmutable, $clock however needs to be of type ClockInterface